### ISSUE TITLE
Panic removing range?

### ISSUE THREAD

**Issue by gsteinLTU** [rlee287/range_union_find#1]
I get this error when doing some calls to remove_range, am I misusing it, or is this a bug? It seemed to be working great otherwise (the inserted range in the example code was originally created by another, successful, call to remove_range).

```
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `0`,
 right: `1`', /Users/gordon/.cargo/registry/src/github.com-1ecc6299db9ec823/range_union_find-0.4.3/src/lib.rs:692:21
```

To recreate:

```
fn main() {
    let mut inv = range_union_find::IntRangeUnionFind::<i32>::new();
    inv.insert_range(&(14..=20));
    inv.remove_range(&(15..=25));
}
```

**Comment by rlee287** [comment]
Sorry for not getting to this until now, but I think this is fixed now in v0.4.4. This particular example has been added as a regression test, but let me know if you encounter any similar issues in the future.

**Comment by gsteinLTU** [comment]
Thanks! This library was really useful for one of the Advent of Code problems (fortunately, knowing which row caused this panic was enough to work with).
